File: practise.py
#Question1
def fizzbuzz(n):
    numbers=range(n)
    for number in numbers:
        if number%3==0 and number%5==0:
            print("FizzBuzz")
        elif number%3==0:
            print("Fizz")
        elif number%5==0:
            print("Buzz")
        else:
            print(number)
def is_Leap(year):
    
    if year%4==0 and year%100!=0:
        if year%400==0:
            print(year, "is a leap year.")
    elif year%4!=0:
        print(year, "is not a leap year.")
print(is_Leap(1900))



def calculate_sum(numbers):
    sum =0
    for num in numbers:
        sum +=num
    print(sum)

calculate_sum([20,2,3,4,10])
    


def check_number_in_list(array, number):
    return number in array

my_list = [1, 2, 3, 4, 5]
number_to_check = 3

if check_number_in_list(my_list, number_to_check):
    print(f"{number_to_check} exists in the list.")
else:
    print(f"{number_to_check} does not exist in the list.")

def is_prime_division(number):
    if number <= 1:
        return False
    if number <= 3:
        return True
    if number % 2 == 0 or number % 3 == 0:
        return False
    i = 5
    while i * i <= number:
        if number % i == 0 or number % (i + 2) == 0:
            return False
        i += 6
    return True

num = 29
if is_prime_division(num):
    print(f"{num} is a prime number.")
else:
    print(f"{num} is not a prime number.")

File: practise.js
//Question1
//input: 
//range of numbers
//Process
//Check if a number is divisible by 3
//check if the number is divisible by 5
//check if the number is both divisible by 3 and 5
//output
//"Fizz"
//"Buzz"
//"FizzBuzz"



function fizzBuzz(limit) {
    for (let i = 1; i <= limit; i++) {
        if (i % 3 === 0 && i % 5 === 0) {
            console.log("FizzBuzz");
        } else if (i % 3 === 0) {
            console.log("Fizz");
        } else if (i % 5 === 0) {
            console.log("Buzz");
        } else {
            console.log(i);
        }
    }
}
fizzBuzz(100);

//Question
//input
//number
//process
//check if the number is a prime number
// Output If its a prime or not

//Question3
function isLeap(year){
    if(year% 400==0)
{
  console.log("leap year")
}
else if(year% 100==0)
{
  console.log("Not a leap year")
}
else if(year % 4==0)
{
  console.log("leap year")
}
else{
  console.log("Not a leap year")
}
}
isLeap(2000)

//Question5
//input
//a character
//process
//check if the character is a vowel
//output
// vowel or Constant

function checkVowel(char) {
    char = char.toLowerCase(); 
    if (char === 'a' || char === 'e' || char === 'i' || char === 'o' || char === 'u') {
        return "Vowel";
    } else {
        return "Consonant";
    }
}
checkVowel('z')

//Question6
//input
//array
//process
//compare the numbers in the array to check whether it is a maximum number or minumum number



//Question7
//input
//array of numbers
//process
//go through the array and add the numbers
//output
//single number of the sumation of the numbers

function calculateSum(numbers){
    sum= numbers.reduce((acc,curr)=>acc+curr)
    console.log(sum)
}
calculateSum([20,2,3,4,10])

//Question8
//input
//array of numbers
//process
//check if the number exist in the array
//statement

function checkNumberInArray(array, number) {
    return array.includes(number);
  }
  
  const myArray = [1, 2, 3, 4, 5];
  const numberToCheck = 3;
  
  if (checkNumberInArray(myArray, numberToCheck)) {
    console.log(`${numberToCheck} exists in the array.`);
  } else {
    console.log(`${numberToCheck} does not exist in the array.`);
  }

  function isPrime(number) {
    if (number < 2) {
      return false;
    }
  
    for (let i = 2; i <= Math.sqrt(number); i++) {
      if (number % i === 0) {
        return false;
      }
    }
  
    return true;// Usage
  }
  
  
  const number = 17;
  console.log(`${number} is prime:`, isPrime(number));
  
